modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java

```diff
@@ -1785,12 +1784,22 @@ private String _getRedirect() {
 	}
 
 	private String _getResourceURL(String resourceID) {
-		ResourceURL resourceURL = renderResponse.createResourceURL();
-
-		resourceURL.setResourceID(resourceID);
-
-		return HttpComponentsUtil.addParameter(
-			resourceURL.toString(), "p_l_mode", Constants.EDIT);
+		return ResourceURLBuilder.createResourceURL(
+			renderResponse.createResourceURL()
+		).setBackURL(
+			ParamUtil.getString(
+				portal.getOriginalServletRequest(httpServletRequest),
+				"p_l_back_url", themeDisplay.getURLCurrent())
+		).setParameter(
+			"backURLTitle",
+			ParamUtil.getString(
+				portal.getOriginalServletRequest(httpServletRequest),
+				"p_l_back_url_title", themeDisplay.getURLCurrent())
```

Comment:
@Mikellorza the title should not be a title? you are setting the current url to the default value

Reply:
You are right. I 've just changed.

thanks @ealonso